go.mod: bump Pebble to afb4d6001bab and update compression metrics

RaduBerinde opened the pull request:
We change the compression metrics from showing sstable counts to
showing compressed bytes and compression ratios.

Changes:

 * [`afb4d600`](https://github.com/cockroachdb/pebble/commit/afb4d600) block: minor change to CompressionRatio()
 * [`d57c0ace`](https://github.com/cockroachdb/pebble/commit/d57c0ace) metrics: move Compression to Table
 * [`750fe8c3`](https://github.com/cockroachdb/pebble/commit/750fe8c3) compression: improve compression stats string
 * [`335af5d4`](https://github.com/cockroachdb/pebble/commit/335af5d4) sstable: free physical buffers on write
 * [`1520f913`](https://github.com/cockroachdb/pebble/commit/1520f913) db: track duration, bytes read for block reads in TableIngestInfo
 * [`503a214b`](https://github.com/cockroachdb/pebble/commit/503a214b) db: track memtable flush, manifest update durations in TableIngestInfo
 * [`c1d36869`](https://github.com/cockroachdb/pebble/commit/c1d36869) metrics: expose new compression stats

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/151315)
<!-- Reviewable:end -->

jbowens reviewed (approved):
lgtm

RaduBerinde committed:
go.mod: bump Pebble to afb4d6001bab and update compression metrics

We change the compression metrics from showing sstable counts to
showing compressed bytes and compression ratios.

Changes:

 * [`afb4d600`](https://github.com/cockroachdb/pebble/commit/afb4d600) block: minor change to CompressionRatio()
 * [`d57c0ace`](https://github.com/cockroachdb/pebble/commit/d57c0ace) metrics: move Compression to Table
 * [`750fe8c3`](https://github.com/cockroachdb/pebble/commit/750fe8c3) compression: improve compression stats string
 * [`335af5d4`](https://github.com/cockroachdb/pebble/commit/335af5d4) sstable: free physical buffers on write
 * [`1520f913`](https://github.com/cockroachdb/pebble/commit/1520f913) db: track duration, bytes read for block reads in TableIngestInfo
 * [`503a214b`](https://github.com/cockroachdb/pebble/commit/503a214b) db: track memtable flush, manifest update durations in TableIngestInfo
 * [`c1d36869`](https://github.com/cockroachdb/pebble/commit/c1d36869) metrics: expose new compression stats

Release note: none.
Epic: none.

RaduBerinde commented:
TFTR!

bors r+
